gnd/src/main/res/layout/map_home_layout.xml

```diff
@@ -64,19 +65,19 @@
       android:layout_width="wrap_content"
       android:layout_height="wrap_content"
       android:onClick="@{() -> viewModel.onLocationLockClick()}"
-      app:srcCompat="@drawable/ic_gps_lock"
-      app:tint="@{viewModel.getIconTint()}"
       app:backgroundTint="@color/colorBackground"
       app:fabSize="normal"
       app:layout_constraintBottom_toTopOf="@+id/add_feature_btn"
       app:layout_constraintEnd_toEndOf="parent"
+      app:srcCompat="@drawable/ic_gps_lock"
+      app:tint="@{viewModel.getIconTint()}"
       app:useCompatPadding="true" />
 
     <com.google.android.material.floatingactionbutton.FloatingActionButton
       android:id="@+id/add_feature_btn"
       android:layout_width="wrap_content"
       android:layout_height="wrap_content"
-      android:visibility="@{homeScreenViewModel.getAddFeatureButtonVisibility()}"
+      android:visibility="@{homeScreenViewModel.isAddFeatureButtonVisible() ? View.VISIBLE : View.GONE}"
```

Comment:
In retrospect we probably should keep all logic inside the view model so that it can be tested easily, not in the layout. I may change this back in future PRs.